Automattic/woocommerce-payments
# Update some workflow files to use an env variable for L-2 versions

Fixes #4685

#### Changes proposed in this Pull Request

Instead of copying/pasting the L-2 value for both WP and WC in several locations for some workflow files, I defined a single variable in the global env of each workflow file to be reused in the job steps.
To do that, I had to implement a step to generate the matrix dynamically, otherwise it was not possible to use an env variable directly.

The goal was to simplify the work of updating the L-2 versions, both locally (current way) or through a script (new way: see the related PR in woorelease: https://github.com/woocommerce/woorelease-extensions/pull/8).

#### Testing instructions

- Trigger the `compatibility.yml` workflow and make sure that all the combinations from the matrix are generated as expected (last run on this branch: https://github.com/Automattic/woocommerce-payments/actions/runs/3311694781; compare with an old run from last week)
- Trigger the `e2e-test.yml` workflow and make sure that all the combinations from the matrix are generated as expected (compare with an old run from last week).
- Trigger the `php-lint-test.yml` workflow and make sure that the WC version installed for the `PHP testing (7.x)` steps is `6.8.2` (i.e. the current L-2 version; you can check for the line `Installing WooCommerce (6.8.2)` in the logs from `Run bash bin/run-ci-tests.bash` step). Last run on this branch: https://github.com/Automattic/woocommerce-payments/actions/runs/3311694813

-------------------


**Post merge**
